code/datums/emotes.dm

```diff
@@ -61,7 +61,13 @@
 		return
 
 	user.log_message(msg, LOG_EMOTE)
-	var/space = should_have_space_before_emote(html_encode(msg)[1]) ? " " : ""
+
+	//Removing space if necessary, and adding punctuation to the end
+	var/space = should_have_space_before_emote(html_encode(msg)) ? " " : ""
+	var/end = copytext(msg, length(msg))
+	if(!(end in list("!", ".", "?", ":", "\"", "-")))
```

Comment:
would add a ' if we have "

Reply:
i'd honestly opt against that; it would probably need to be copied to the say code, and in that case typing a message like, say, [that hat is Beans'], you still want a period added at the end. also i don't even think the " not adding a period works for the same reason regex catching ' is such a motherfucker and is the base reason this pr exists. tl;dr withered emoji i'll glance at it i guess

Reply:
![image](https://user-images.githubusercontent.com/10237657/173458006-73d688bb-9055-4673-b8c4-3aa8491a01b5.png)
i guess it is time to brick everything by trying to refactor the end-punctuation detection for automatically adding a period. please kill me